src/main/java/com/minecolonies/coremod/commands/AbstractMenu.java

```diff
@@ -23,4 +23,61 @@ public void setMenuType(final IMenuType menuType)
     {
         this.menuType = menuType;
     }
+
+    @Override
+    public int hashCode()
+    {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((menuItemName == null) ? 0 : menuItemName.hashCode());
+        result = prime * result + ((menuType == null) ? 0 : menuType.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(final Object obj)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 8, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)